### PR TITLE
Document templates in-place

### PR DESCRIPTION
**Pull request by ingalless**
Relates to: https://github.com/foambubble/foam/pull/359

### CODE REVIEW

File: .foam/templates/your-first-template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Foam Note Templates
+
+Foam includes note templates! This allows you to create new notes from the given template.
```
Comment (riccardoferretti) [inline .foam/templates/your-first-template.md]:
```suggestion
Foam includes note templates! 
This allows you to easily create notes that have similar structure without having to use copy/paste :)

```

File: .foam/templates/your-first-template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+1. ${1:First tabstop}
+2. ${2:A second tabstop}
+3. ${3:A third tabstop}
+
+Note Created: ${CURRENT_YEAR}${CURRENT_MONTH}${CURRENT_DATE}
+
+---
+
+Try out the above example by running the `Foam: Create New Note From Template` command and selecting the `your-first-template` template. Notice what happens when your new note is created!
```
Comment (riccardoferretti) [inline .foam/templates/your-first-template.md:21]:
```suggestion
Try out the above example by running the `Foam: Create New Note From Template` command and selecting the `your-first-template` template. Notice what happens when your new note is created!

To remove this template, simply delete the `.foam/templates/your-first-template.md` file.

Enjoy!
```

File: .foam/templates/your-first-template.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Foam Note Templates
+
+Foam includes note templates! This allows you to create new notes from the given template.
+Templates are treated just like snippets allowing you to use [VS Code's Snippet Syntax](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax).
```
Comment (riccardoferretti) [inline .foam/templates/your-first-template.md]:
```suggestion
Templates support the [VS Code's Snippet Syntax](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax), which means you can:
- add variables to the newly created note
- add tabstop to automatically navigate to the key parts of the note, just like a form
```

**Review by riccardoferretti — APPROVED** [review]
Looks good!
Expanded a bit on your text, let me know what you think

**Comment by ingalless** [comment]
@riccardoferretti Good suggestions! Applied 👍🏻

**Pull request merged**